dev-tools/make/mage.mk

```diff
@@ -1,13 +1,69 @@
-MAGE_VERSION     ?= v1.9.0
-MAGE_PRESENT     := $(shell mage --version 2> /dev/null | grep $(MAGE_VERSION))
-MAGE_IMPORT_PATH ?= github.com/magefile/mage
-export MAGE_IMPORT_PATH
-
-.PHONY: mage
-mage:
-ifndef MAGE_PRESENT
-	@echo Installing mage $(MAGE_VERSION) from vendor dir.
-	@go install -mod=vendor -ldflags="-X $(MAGE_IMPORT_PATH)/mage.gitTag=$(MAGE_VERSION)" ${MAGE_IMPORT_PATH}/...
-	@-mage -clean
-endif
-	@true
+# This is a minimal Makefile for Beats that are built with Mage. Its only
+# responsibility is to provide compatibility with existing Jenkins and Travis
+# setups.
+
+#
+# Variables
+#
+.DEFAULT_GOAL := help
+PWD           := $(CURDIR)
+
+#
+# Includes
+#
+include $(ES_BEATS)/dev-tools/make/mage-install.mk
+
+#
+# Targets (alphabetically sorted).
+#
+.PHONY: check
+check: mage
+	mage check
+
+.PHONY: clean
+clean: mage
+	mage clean
+
+fix-permissions:
+
+.PHONY: fmt
+fmt: mage
+	mage fmt
+
+# Default target.
+.PHONY: help
+help:
+	@echo Use mage rather than make. Here are the available mage targets:
+	@mage -l
+
+.PHONY: release
+release: mage
+	mage package
+
+stop-environment:
+
+.PHONY: unit-tests
+unit-tests: mage
+	mage update unitTest
```

Comment:
Since this is a new target, I don't think `update` should be required. The unitTest targets should have the appropriate dependencies wired up already. In most cases I think they do now, like requiring the `Fields` before running the python tests.

Reply:
The diff is wierd, because this is actually just `xpack.mk` that was renamed to `mage.mk` because now its not only x-pack projects that use the file.

I will remove it but it does affect, all x-pack projects. Hopefully those do the correct thing.